sql/ts/src/skdb_util.ts

```diff
@@ -102,3 +104,42 @@ export class SKDBTable extends Array<Record<string, any>> {
     });
   }
 }
+
+/* ***************************************************************************/
+/* Transaction API for programmatically constructing and executing
+   transactions against an SKDB instance */
+/* ***************************************************************************/
+
+export class SKDBTransaction {
+  private query_params: Params;
+  private stmts: Array<string>;
+  private db_handle: SKDB;
+
+  constructor(skdb: SKDB) {
+    this.query_params = {};
+    this.stmts = [];
+    this.db_handle = skdb;
+  }
+
+  addStatement(stmt: string): SKDBTransaction {
+    this.stmts.push(stmt.trim().replace(/;$/, ""));
+    return this;
+  }
+
+  params(params: Params): SKDBTransaction {
+    Object.assign(this.query_params, params);
+    return this;
+  }
+
+  flush(params: Params = {}): Promise<SKDBTable> {
```

Comment:
Let's call this 'commit'? Just to be inline with usual txn keywords.

Reply:
The params 'set union' behaviour and the argument to flush unioning is a little unexpected imo, but desirable. Maybe we could call the method 'addParams' and the arg here 'additionalParams' or something like that? I read 'params()' as replacement.

I made the opposite mistake to this with `mirror()` and I regret it. I wish mirror were called 'mirrorOnly' or something like that. The mirror verb suggests being additive to me.